### ISSUE TITLE
Somethings wrong with Unet or MANet decoder when output_stride != 32

### ISSUE THREAD

**Issue by dulicui742** [qubvel/segmentation_models.pytorch#749]
class DecoderBlock:
...
def forward(self, x, skip=None):
          **x = F.interpolate(x, scale_factor=2, mode="nearest")**
          if skip is not None:
              x = torch.cat([x, skip], dim=1)
              x = self.attention1(x)
          x = self.conv1(x)
          x = self.conv2(x)
          x = self.attention2(x)
          return x

when use dialated conv, the output is 8x or 16x, the begining concat operation don't need to upsample!

**Comment by qubvel** [comment]
Hi, Unet were not designed to work with dilated encoders
